src/expr/src/expr/expr_some_all.rs

```diff
@@ -110,10 +110,11 @@ impl Expression for SomeAllExpression {
                 let datum_right = right.unwrap();
                 match datum_right {
                     ScalarRefImpl::List(array) => {
-                        let len = array.iter().len();
+                        let flattened = array.flatten();
+                        let len = flattened.len();
```

Comment:
Are we going to support `1 = any(array[array[1,2], array[3,4]])` ? cc @xiangjinwu 

```
dev=> select 1 = any(array[array[1,2], array[3,4]]);
ERROR:  QueryError: Bind error: failed to bind expression: 1 = SOME(ARRAY[ARRAY[1, 2], ARRAY[3, 4]])

Caused by:
  Feature is not yet implemented: Equal[Int32, List]
Tracking issue: https://github.com/risingwavelabs/risingwave/issues/112
```